spec/migrations/add_from_template_permissions_to_roles_with_add_permissions_spec.rb

```diff
@@ -0,0 +1,213 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+require "spec_helper"
+require Rails.root.join("db/migrate/20251113124140_add_from_template_permissions_to_roles_with_add_permissions")
+
+RSpec.describe AddFromTemplatePermissionsToRolesWithAddPermissions, type: :model do
+  # Silencing migration logs, since we are not interested in that during testing
+  subject(:run_migration) { ActiveRecord::Migration.suppress_messages { described_class.migrate(:up) } }
```

Comment:
Can be converted to a method to allow idempotence test (and extract `direction` to reuse in `:down` test):

```suggestion
  let(:direction) { :up }

  def run_migration
    # Silencing migration logs, since we are not interested in that during testing
    ActiveRecord::Migration.suppress_messages { described_class.migrate(direction) }
  end
```